keep-ui/app/(keep)/workflows/create-workflow-modal/use-query-workflow-template.ts

```diff
@@ -0,0 +1,35 @@
+import { WorkflowTemplate } from "@/shared/api/workflows";
+import { useApi } from "@/shared/lib/hooks/useApi";
+import useSWR, { SWRConfiguration } from "swr";
+
+export interface WorkflowTemplatesQuery {
+  cel: string;
+  limit: number;
+  offset: number;
+}
+
+export function useQueryWorkflowTemplate(
+  query: WorkflowTemplatesQuery,
+  options?: SWRConfiguration<any>
+) {
+  const api = useApi();
+  const requestUrl = `/workflows/templates/query`;
+  const { data, error, isLoading, mutate } = useSWR<any>(
+    api.isReady() && query
```

Comment:
also, here can you use the patter from https://github.com/keephq/keep/blob/main/keep-ui/entities/workflows/model/workflowKeys.ts for keys